src/server/frontend_wayland/xdg_decoration_unstable_v1.cpp

```diff
@@ -47,10 +46,13 @@ class ToplevelsWithDecorations
         return inserted;
     }
 
-    /// \return true if only one element was erased, false otherwise.
+    /// \return true if the toplevel didn't exist in the set, false otherwise.
+    /// The return value is used in the destroy callback of the toplevel to
+    /// check if the toplevel is destroyed before the decoration (orphaned
+    /// decoration).
```

Comment:
1. This is a very weird semantic: returning `false` on success.
2. It is normal to document the preconditions, action, and postconditions of a function. Not where its result happens to be used

Reply:
I agree. I myself was a bit confused while using this method. 

I've moved the explanation to where the mentioned case is actually handled. I might've detailed things too much, but I think this will save the next person (likely me :)) that works on this code some time.